src/test/java/tests/BaseMenuTestClass.java

```diff
@@ -0,0 +1,205 @@
+                .thenReturn(FileUtils.getFile(this.getClass(), "test_restore.xml"));
+        setupInstallServiceMock();
+    }
+
+    private String resolveAppId(String ref){
+        log.info("Test resolving hack ref: " + ref);
+        String appId = ref.substring(ref.indexOf("app_id=") + "app_id=".length(),
+                ref.indexOf("#hack"));
+        log.info("Got appId: " + appId);
+        switch (appId) {
+            case "doublemgmtappid":
+                ref = "archives/parent_child.ccz";
+                break;
+            case "navigatorappid":
+                ref = "archives/parent_child.ccz";
+                break;
+            case "caseappid":
+                ref = "apps/basic2/profile.ccpr";
+                break;
+            case "createappid":
+                ref = "archives/basic.ccz";
+                break;
+            case "casemediaappid":
+                ref = "archives/casemedia.ccz";
+                break;
+            case "endformappid":
+                ref = "archives/formnav.ccz";
+                break;
+            case "langsappid":
+                ref = "archives/langs.ccz";
+                break;
+            case "casetilesappid":
+                ref = "archives/casetiles.ccz";
+                break;
+            default:
+                throw new RuntimeException("Couldn't resolve appId for ref: " + ref);
+        }
+        log.info("Resolved ref: " + ref);
+        return ref;
+    }
+
+    private void setupInstallServiceMock() {
+        try {
+            doAnswer(new Answer<Object>() {
+                @Override
+                public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                    try {
+                        Object[] args = invocationOnMock.getArguments();
+                        String ref = (String) args[0];
+                        if(ref.contains("#hack=commcare.ccz")){
+                            ref = resolveAppId(ref);
+                        }
+                        String username = (String) args[1];
+                        String path = (String) args[2];
+                        FormplayerConfigEngine engine = new FormplayerConfigEngine(username, path);
+                        String absolutePath = getTestResourcePath(ref);
+                        System.out.println("Init with path: " + absolutePath);
+                        if (absolutePath.endsWith(".ccpr")) {
+                            engine.initFromLocalFileResource(absolutePath);
+                        } else if (absolutePath.endsWith(".ccz")) {
+                            engine.initFromArchive(absolutePath);
+                        } else {
+                            throw new RuntimeException("Can't install with reference: " + absolutePath);
+                        }
+                        engine.initEnvironment();
+                        return engine;
+                    } catch (Exception e) {
+                        e.printStackTrace();
```

Comment:
doesn't hte stack trace get printed and halt execution anyways? seems like it'd have the same behavior if we removed this try block